### PR TITLE
PR: Add option to connect to external PyLS servers

### LINKED ISSUE #9195: Cannot connect to an external PyLS server
## Problem Description

The new menu does not allow to connect to an external PyLS server. 

### What steps reproduce the problem?

1. Run spyder-pyls server on one terminal with the command `python -m pyls -vvv --port 2089 --tcp`
2. Run spyder on another terminal
3. Try to connect with the running server by changing the port in Preferences > Completion and linting > Advanced 
4. After giving apply, there's no new connection received by the local pyls server.
* The same happens after trying to setup a new server for python.

### What is the expected output? What do you see instead?
Connect to an external PyLS server running locally.

## Versions

* Spyder version: Spyder 4.0.0.dev0 (Commit: ba26876de) 
* Python version: 3.6.8
* Qt version: 5.12.2
* PyQt version: 5.12.1 
* Operating System name/version: Darwin 18.2.0 

### Dependencies

```
pygments >=2.0    :  2.3.1 (OK)
qdarkstyle >=2.6.4:  2.6.5 (OK)
sphinx >=0.6.6    :  1.8.5 (OK)
psutil >=0.3      :  5.6.1 (OK)
pyls >=0.19.0     :  0.21.5 (OK)
nbconvert >=4.0   :  5.4.1 (OK)
pandas >=0.13.1   :  None (NOK)
numpy >=1.7       :  1.16.2 (OK)
sympy >=0.7.3     :  None (NOK)
cython >=0.21     :  None (NOK)
qtconsole >=4.2.0 :  4.4.3 (OK)
IPython >=4.0     :  7.4.0 (OK)
matplotlib >=2.0.0:  None (NOK)
pylint >=0.25     :  2.3.1 (OK)
```

### PR DESCRIPTION
**Pull request by dalthviz**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)
* [x] Included a screenshot (if affecting the UI)

imagen

Properly set the `external` property for the lsp-server

### Issue(s) Resolved

Fixes #9195 

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: dalthviz

### CODE REVIEW

File: spyder/plugins/editor/lsp/manager.py [hunk 1]
```diff
@@ -82,7 +82,7 @@ def get_language_config(self, language):
         """Get language configuration options from our config system."""
         if language == 'python':
             self.update_python_config()
-            return self.python_config
+            return self.python_config.copy()
```
Comment (ccordoba12) [inline spyder/plugins/editor/lsp/manager.py]: Why is this needed? I mean, it shouldn't be necessary.

File: spyder/plugins/editor/lsp/manager.py [hunk 2]
```diff
@@ -301,14 +301,16 @@ def update_python_config(self):
             'follow_imports': self.get_option('jedi_definition/follow_imports')
         }
 
+        # Advanced
+        advanced_external = self.get_option('advanced/external')
```
Comment (ccordoba12) [inline spyder/plugins/editor/lsp/manager.py]: Let's change this to

    external_server = self.get_option('advanced/external')

File: spyder/plugins/editor/lsp/manager.py [hunk 3]
```diff
@@ -301,14 +301,16 @@ def update_python_config(self):
...
+
         # Setup options in json
         self.python_config['cmd'] = cmd
         if host in self.LOCALHOST:
             self.python_config['args'] = '--host {host} --port {port} --tcp'
-            self.python_config['external'] = False
         else:
             self.python_config['args'] = ''
-            self.python_config['external'] = True
+        self.python_config['external'] = advanced_external
```
Comment (goanpeca) [inline spyder/plugins/editor/lsp/manager.py]: This line alone is the actual fix?

File: spyder/plugins/editor/lsp/manager.py (hunk 1 above)
Comment (dalthviz) [reply spyder/plugins/editor/lsp/manager.py]: While testing the changes, the restart was not being triggered (the `current_config` and `new_config` in `update_server_list` were the same). This was caused because the client for `python` was being affected for the change in `self.python_config` (adding the `copy` prevents this to happen)

File: spyder/plugins/editor/lsp/manager.py (hunk 3 above)
Comment (dalthviz) [reply spyder/plugins/editor/lsp/manager.py]: Yep, in this way we can handle an pyls instant launched outside Spyder (even if it's launched in the same machine where Spyder is running and besides the error in the restart of the client)

File: spyder/plugins/editor/lsp/manager.py (hunk 1 above)
Comment (ccordoba12) [reply spyder/plugins/editor/lsp/manager.py]: Ok, I understand what you mean now and that's a serious problem because by using a shared `self.python_config` attribute we are modifying the configuration of the existing Python client and of future clients too.

To avoid that, please do the following:

1. Remove `self.python_config` from `LSPManager`
2. Rename `update_python_config` to `generate_python_config`
3. Use a local variable inside that method called `python_config`, like this:

    ```python
    def generate_python_config(self):
    """
    Generate Python server configuration with the options saved in our
    config system
    """
    python_config = PYTHON_CONFIG.copy()
    ...
    ...
    return python_config
    ```

    and use `python_config` instead of `self.python_config` in that method.

File: spyder/plugins/editor/lsp/manager.py (hunk 2 above)
Comment (ccordoba12) [reply spyder/plugins/editor/lsp/manager.py]: You changed the names of other variables, but forgot to change precisely this name ;-)

**Comment by ccordoba12** [comment]
@dalthviz, did you test that my last suggestion works as expected?

**Comment by dalthviz** [comment]
@ccordoba12 yep, it works for me :+1:
